### ISSUE TITLE
WorkAdventure Accessibility Issues

### ISSUE THREAD

**Issue by JamesGable** [thecodingmachine/workadventure#1055]
Adding my comments in hope that accessibility issues with WorkAdventure get fixed. Any group which receives U.S. federal funding, such as many schools, libraries, and airports, and others cannot use this software until these accessibility issues are fixed. 

**CRITICAL ISSUES**
- Buttons must have discernible text
- Images must have alternate text
- Form elements must have labels
- Select element must have an accessible name
- Zooming and scaling should not be disabled

**MODERATE ISSUES**
- Ensure that scrollable region has keyboard access
- Heading levels should only increase by one
- Document should not have more than one main landmark
- Ensures landmarks are unique
- All page content should be contained by landmarks

**Comment by jonnytest1** [comment]
🤔 while the general ui could get made accessible i dont see how you intent to implement accessible maps 
those are not drawn in the html but rendered onto a canvas

**Comment by moufmouf** [comment]
Hey @JamesGable ,

Are you referring to the website (workadventu.re) or to the "game" (play.workadventu.re)?

Critical issues should be quite easy to fix (and will be fixed soon), but will it really make WorkAdventure accessible? As @jonnytest1 says quite correctly, WorkAdventure draws the map on a canvas which cannot be read by a screen reader for blind people. This should certainly help visually impaired people though.

**Comment by Jacques-Olivier-Farcy** [comment]
Hi,

I think it is a good point @JamesGable. Even if it will not worked for every handicap, there is a real need to make the WorkAdventure game more accessible.
I hope for example that Jitsi will be capable of transcript audio to text.

As i am intersted to work on this aspect, @JamesGable, do you have any references on the recommendations for managing accessibility?

**Comment by jonnytest1** [comment]
> refe

i think jitsi already has a transcript mode of some sort
it may be required to enable it in the jitisConfig or jitsiInterfaceConfig properties

**Comment by tabascoeye** [comment]
looks to be a bit more setup on the jitsi server itself needed as well
https://community.jitsi.org/t/jitsi-transcription/88562

**Comment by kloh-fr** [comment]
The whole map experience is clearly almost impossible to make accessible for every users (as mentioned, screen reader users would have quite some difficulties in any case).

Yet, it would be possible to think of alternative ways of interaction/navigation that would allow such user to enjoy the experience on their own.\
I think about adding some navigation options to show the list of points of interest or the list of people on the map for instance. From these kind of menus, a screen reader user could navigate through them, find the point of interest he/she wants to go to (or person he/she wants to discuss with), choose it and get teleported to the right place/character on the map. Kind of a teleportation function in some way.

By the way, I'm pretty sure this kind of functionnality would be helpful for other people too.\
I indeed think that people who are not familiar with video games can probably be lost in the map experience too. So having some alternative navigations that could be displayed in complement of the map would help them too.

**Comment by behofer** [comment]
I'm a screenreader user myself and I'd really love to see improvements when it comes to navigation. I'd second @kloh-fr's post! Is there any progress going on regarding this?

**Comment by Schroedingberg** [comment]
I'm currently working with a cultural institution who work with WA and need to increase accessibility, especially for visually impaired people. We might have to customize WA for that purpose. If that happens, the best case scenario would be to contribute the results back to upstream. Anyone interested, please get in touch, I really want to avoid overlooking anything :)

**Comment by ValdoTR** [comment]
Hi, some updates on this topic:

**Accessibility**
So like everyone here understand, the platform currently does not prioritize accessibility because it is designed as a gaming engine and is not optimized for accessibility. We are considering integrating Matrix technology for a chat-only experience, which might enhance accessibility for individuals with disabilities. However, this enhancement is not expected to be implemented until at least six months from now.

**Captioning for Events**
If you are organizing an event with a conference and require captions, the recommended approach is to use live streaming on YouTube and embed the live stream on WorkAdventure. YouTube's auto-captioning feature can then be utilized for accessibility. Jitsi captions are not mature yet to be implemented.

**Mobile and Tablet Readability**
It's important to note that readability on mobile and tablet devices is and will always be less effective due to smaller screens. The need to display the chat, game interface, and potentially a website simultaneously is just too much and in order to have a really good experience on mobile, we would need to develop a brand new mobile app with a completely different UI. If you aim for a mobile-first experience, be cautious about the size of maps for performance reasons.

@JamesGable is the experience better now? If you still have the same issues please UP the issue here.
@kloh-fr since your post, we added a feature that displays the list of connected users, and the capability to "walk to" any user in that list.
@Schroedingberg can you provide more info on that? it would be super cool to contribute if you can. You can ask people if they are interested to help you in our Discord server: https://discord.gg/G6Xh9ZM9aR

**Comment by patcon** [comment]
Thanks for the context @ValdoTR 

> I think about adding some navigation options to show the list of points of interest or the list of people on the map for instance. From these kind of menus, a screen reader user could navigate through them, find the point of interest he/she wants to go to (or person he/she wants to discuss with), choose it and get teleported to the right place/character on the map. Kind of a teleportation function in some way.

Really appreciate your idea @kloh-fr -- I feel like a totally alternate "view" is possible, that simple understands the substance of WA as a bunch of ephemeral video chat rooms that one can hop between participating in.
- this accessible view would have a list of all rooms to select from, and use common a11y patterns to push state change info (who is in what rooms, or counts of people in rooms) to the browser as a feed. "user patcon joined 
- each person on their own (when not within range of someone) is in their own "room"
- when two people get together, a new randomly named room is created (e.g., "tidy-trot-starling" or some random word set) and both people enter
- any new arrivals to that cluster leave their individual rooms and enter the group room
- some logic for when groups join or depart the larger group, as new groups would need to be instantiated

now, a person with visual impairment can see a list of video chat rooms without worrying about spatial navigation. some features would be missing, but at least the most important piece (the other ppl) would be available to them.

I am unfamiliar with the internals, but can imagine that with the right API interfaces or websocket push affordances exposed, someone could build this logic as an extension, and have it exist separately from the main work-adventure project <3

Anyhow, love this project! Hopefully the above brainstorm is in some way helpful 🙏
